dts/bindings/clock/atmel,sam0-gclk.yaml

```diff
@@ -0,0 +1,30 @@
+#
+# Copyright (c) 2019 Derek Hageman <hageman@inthat.cloud>
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+title: SAM0 Generic Clock (GCLK) provider
+version: 0.1
+
+description: >
+    This is a representation of a SAM0 Generic Clock provider.  A GCLK is
+    a clock distribution mechanism that connects an input clock with various
+    consumers.  It can optionally provide a divisor on the input clock, so
+    the output clock frequency may be different than the input frequency.
+
+inherits:
+    !include clock.yaml
+
+properties:
+    compatible:
+      constraint: "atmel,sam0-gclk"
+
+    clock-frequency:
```

Comment:
What is `clock-frequency` suppose to mean?

Reply:
This is the output clock frequency of the GCLK.  This is required because it is used to calculate the divisor the GCLK applies to its input and because knowing the input frequency would require a recursive walk otherwise (i.e. when a GCLK sources from another GCLK).

I have clarified the description on the binding.